release/models/system/openconfig-system.yang

```diff
@@ -371,6 +377,14 @@ module openconfig-system {
           "The current system date and time.";
     }
 
+    leaf up-time {
+        type oc-types:timeticks64;
+        units "nanoseconds";
```

Comment:
This comment can apply to other leafs that are tagged with `units nanoseconds` but would like to clarify the expectations around time precision

1. To accurately represent precision as nanoseconds
2. To represent solely according to the datatype regardless of actual precision

I think what we'd probably find if we audited all implementations across all such leafs, the internal representation may likely be less precise and there will just be a conversion to fulfill the datatype in various cases.  It may give the perception of nanoseconds but may infact just be seconds/centiseconds, etc..

And if we look across the vendor references listed here, it is likely we'd actually find seconds (or less) precision today (across common show commands)

For Linux systems, if `/proc/uptime` is leveraged, kernel sources currently convert this to centiseconds

Are there any vendor or common OSS src references to nanoseconds precision?  Otherwise it seems to fulfill the accurate precision requested here (which I'd be curious on how nanoseconds on uptime would actually be used), it would likely be all new work for each implementation

Reply:
Also note that the `timeticks64` type defined in `openconfig-types` already hard codes a unit of `nanoseconds` so it is implied that any leaf using that type inherits that unit (this might need revisiting outside of this PR unless we agree its about presentation and not precision)

Quick output from a shipping implementation will yield seconds precision packed into a timeticks64 type dropping the tail `648933887`

```
$ date -d @1704321883
Wed Jan  3 15:44:43 MST 2024
$ date -d @1704321916
Wed Jan  3 15:45:16 MST 2024
```

```
system/processes/process[pid=1736]/state/start-time: 1704321883648933887
system/processes/process[pid=2243]/state/start-time: 1704321916648933887
system/processes/process[pid=1947]/state/start-time: 1704321903648933887
system/processes/process[pid=40]/state/start-time: 1704321850648933887
system/processes/process[pid=1729]/state/start-time: 1704321883648933887
system/processes/process[pid=1945]/state/start-time: 1704321903648933887
system/processes/process[pid=1153]/state/start-time: 1704321855648933887
system/processes/process[pid=1391]/state/start-time: 1704321857648933887
system/processes/process[pid=42]/state/start-time: 1704321850648933887
system/processes/process[pid=1378]/state/start-time: 1704321857648933887
system/processes/process[pid=1085]/state/start-time: 1704321855648933887
system/processes/process[pid=1404]/state/start-time: 1704321857648933887
system/processes/process[pid=1755]/state/start-time: 1704321883648933887
system/processes/process[pid=1472]/state/start-time: 1704321858648933887
system/processes/process[pid=1086]/state/start-time: 1704321855648933887
system/processes/process[pid=1944]/state/start-time: 1704321903648933887
```

Reply:
I think we can always say an implementation must send data according to the datatype, even if the implementation cannot support the precision implied.  

There is not a strong operational requirement to maintain uptime with nanosecond precision.  I used nanoseconds to be consistent with the other system time related leafs like boottime and last-configuration-timestamp

Reply:
I'll also note that currently OC only has a timeticks64 type we've used throughout the models.  I think if we want to assert precision requirements we'll need a separate effort for this.